tests/difficulty_test_util.go

```diff
@@ -57,7 +59,10 @@ func (test *DifficultyTest) Run(config *params.ChainConfig) error {
 		UncleHash:  test.UncleHash,
 	}
 
-	actual := ethash.CalcDifficulty(config, test.CurrentTimestamp.Uint64(), parent)
+	chain, _ := core.NewBlockChain(nil, config, nil, vm.Config{})
+	defer chain.Stop()
+
+	actual := ethash.NewFaker().CalcDifficulty(chain, test.CurrentTimestamp.Uint64(), parent)
```

Comment:
This became a bit convoluted. I think it's fine to retain the global method `ethash.CalcDifficulty`, and just wrap it with the `engine.CalcDifficulty`.